src/component/project/ProjectList/ProjectList.tsx

```diff
@@ -43,14 +43,22 @@ function resolveCreateButtonData(isOss: boolean, hasAccess: boolean) {
     }
 }
 
-const ProjectListNew = () => {
+export const ProjectListNew = () => {
     const { hasAccess } = useContext(AccessContext);
     const history = useHistory();
     const styles = useStyles();
     const { projects, loading, error, refetch } = useProjects();
     const [fetchedProjects, setFetchedProjects] = useState<projectMap>({});
     const ref = useLoading(loading);
     const { isOss } = useUiConfig();
+    const [filter, setFilter] = useState('');
+
+    const filteredProjects = useMemo(() => {
+        const regExp = new RegExp(filter, 'i');
+        return filter
+            ? projects?.filter(project => regExp.test(project?.name))
```

Comment:
Looks like `projects` will always be defined, so we don't need the `?`. Same for `project?.name`.